app/routes.py

```diff
@@ -0,0 +1,193 @@
+        if form.validate():
+            episode = Episode(
+                name=form.title.data,
+                user_id=current_user.get_id()
+            )
+            upload_folder = app.config['UPLOAD_PODCAST_FOLDER']
+            static_path = app.config['MEDIA_ROOT']
+            path = f'{static_path}{upload_folder}'
+            if not os.path.exists(path):
+                os.mkdir(path)
+            db.session.add(episode)
+            db.session.flush()
+            db.session.commit()
+            episode_path = os.path.join(path, f'{episode.id}.mp3')
+            form.file.data.save(episode_path)
+            episode.generate_wrapped_file(episode_path)
+    return render_template('upload_podcast.html', form=form, feed_blank='Podcast Main page: RSS feed')
+
+
+@app.route('/user/<username>')
+@login_required
+def profile(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    episodes = Episode.query.filter_by(user_id=user.id)
+    jokes = Joke.query
+    return render_template('profile.html', user=user, joks=jokes, episodes=episodes)
+
+
+@app.route('/jokes/edit_joke/<joke_id>', methods=['GET', 'POST'])
+def edit_joke(joke_id):
+    form = EditJokeForm()
+    joke = Joke.query.filter_by(id=joke_id).first_or_404()
+    if form.validate_on_submit():
+        joke.joke_text = form.text.data
+        db.session.add(joke)
+        db.session.commit()
+
+        # Generating audio file for joke
+        # TODO: add to background queue when ready
+        joke.generate_wrapped_file()
+
+        flash('Ваши изменения сохранены!')
+        return redirect(url_for('profile', username=current_user.username))
+    elif request.method == "GET":
+        form.text.data = joke.joke_text
+    return render_template('edit_joke.html', form=form)
+
+
+@app.route('/jokes/delete_joke/<joke_id>', methods=['GET', 'POST'])
+def delete_joke(joke_id):
+    joke = Joke.query.filter_by(id=joke_id).first_or_404()
+    db.session.delete(joke)
+    db.session.commit()
+    flash("Успешно удалено")
+    return redirect(url_for('profile', username=current_user.username))
+
+
+@app.route('/user/edit_profile/<username>', methods=['GET', 'POST'])
+def edit_profile(username):
+    form = EditUserProfileForm()
+    user = User.query.filter_by(username=username).first_or_404()
+    if form.validate_on_submit():
+        if current_user.id == user.id:
+            if user.check_password(form.old_password.data):
+                user.username = form.username.data
+                user.set_password(form.password.data)
+                db.session.add(user)
+                db.session.commit()
+                flash('Ваши изменения сохранены!')
+                return redirect(url_for('profile', username=user.username))
+            else:
+                flash("Старый пароль неверный")
+                return redirect(url_for("/user/edit_profile/", username=user.username))
+        else:
+            flash('Доступ запрещён')
```

Comment:
Validation of conditions that can be validated on the front end need to happen there and prevent requests to the back end under normal circumstances; therefore on the back end, there is no need to distinguish between different types of validation conditions.